zone/client_process.cpp

```diff
@@ -289,6 +289,27 @@ bool Client::Process() {
 			entity_list.ScanCloseMobs(close_mobs, this, IsMoving());
 		}
 
+        if (RuleB(Inventory, LazyLoadBank) && Connected()) {
+			uint32 banker_max_dist = 0;
+			if(sent_inventory <= EQ::invslot::SHARED_BANK_END && entity_list.GetClosestBanker(this, banker_max_dist) && banker_max_dist <= USE_NPC_RANGE2) {
+				const EQ::ItemInstance* inst = nullptr;
+
+				// Jump the gaps
+				if (sent_inventory < EQ::invslot::BANK_BEGIN) {
+					sent_inventory = EQ::invslot::BANK_BEGIN;
+				} else if (sent_inventory > EQ::invslot::BANK_END && sent_inventory < EQ::invslot::SHARED_BANK_BEGIN) {
+					sent_inventory = EQ::invslot::SHARED_BANK_BEGIN;
+				} else {
+					sent_inventory++;
```

Comment:
So this logic very heavily relies on us hitting Client::Process hundreds of cycles to send the full inventory. Why don't we just do a for loop in the logic and range up to `EQ::invslot::SHARED_BANK_END` once we hit this logic and send the item packet?

Reply:
This is assuming that you have 200 slot bags. I had test cases where I could crash the client by sending anywhere close to that many items updates. Rate limiting it even a little bit (1bag&contents per frame), I wasn't able to duplicate the client crashes.

Reply:
I'm talking even today we have to assume theoretical maximum.

Bank = 8 slots * 3 columns = 24 top slots. 24 top slots by bag slots = 240

Makes sense on the crash issue, normally we bulk send in one packet on zone in and now with lazy loading we could potentially send 240 individual packets (pre 200 slot bags).

Were your client crashes influenced by MQ or was it happening standalone?